cylc/flow/clean.py

```diff
@@ -358,6 +362,10 @@ def remote_clean(
         raise PlatformLookupError(
             f"Cannot clean {id_} on remote platforms as the workflow database "
             f"is out of date/inconsistent with the global config - {exc}")
+
+    with suppress(ISO8601SyntaxError):
+        timeout = str(DurationParser().parse(timeout).get_seconds())
```

Comment:
Shouldn't this error be raised rather than suppressed?

Reply:
I did this to allow users to still use a raw number of seconds. We weren't doing any validation on that before, I suppose your suggestion is a good idea

Reply:
I'm not sure that it's reasonable to use two try excepts here because we can't say for sure which way of setting this value will become the popular one with users:

IMO

```python
if timeout.isdigit():
    timeout = int(timeout)
else:
    try:
        timeout = str(DurationParser().parse(timeout).get_seconds())
    except ISO8601SyntaxError:
        raise InputError(...)
    
```

Reply:
Jupyter notebook testing this.

```python
from metomi.isodatetime.parsers import DurationParser
from metomi.isodatetime.exceptions import ISO8601SyntaxError

def one(timeout):
    try: 
        timeout = str(DurationParser().parse(timeout).get_seconds())
    except ISO8601SyntaxError:
        try:
            timeout = int(timeout)
        except ValueError:
            raise InputError("Hi")


def two(timeout):
    if timeout.isdigit():
        timeout = int(timeout)
    else:
        try:
            timeout = str(DurationParser().parse(timeout).get_seconds())
        except ISO8601SyntaxError:
            raise InputError(...)
        
```


```python
from random import randint, choice
from contextlib import suppress
from cylc.flow.exceptions import InputError
```


```python
%%timeit
one(choice([str(randint(0, 3600))]))
```

    3.53 µs ± 34.1 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)



```python
%%timeit
two(choice([str(randint(0, 3600))]))
```

    1.1 µs ± 13 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)



```python
%%timeit
one(choice([str(randint(0, 3600)), 'PT4S']))
```

    7.61 µs ± 85.8 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)



```python
%%timeit
two(choice([str(randint(0, 3600)), 'PT4S']))
```

    6.23 µs ± 143 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)



```python
%%timeit
with suppress(InputError):
    one(choice([str(randint(0, 3600)), 'PT4S', 'garbage']))
```

    7.76 µs ± 180 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)



```python
%%timeit
with suppress(InputError):
    two(choice([str(randint(0, 3600)), 'PT4S', 'garbage']))
```

    6.39 µs ± 174 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

Reply:
I've made the default `PT5M` so I suspect the ISO branch will be hit most of the time. Anyway, a few ms shouldn't matter too much as this will only be executed once per clean (I'll move this validation to `scripts/clean.py`)